### PR TITLE
Upgrading to spring boot 2.1.2

### PR DESCRIPTION
**Pull request by KevinJCross**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RPE-908
### Change description ###
upgrade to spring boot 2

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -85,72 +46,16 @@ sourceSets {
...
-  compile group: 'io.springfox', name: 'springfox-swagger-ui', version: versions.springfoxSwagger
-
-  compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind'
-  compile group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310'
-
-  compile group: 'uk.gov.hmcts.reform', name: 'java-logging-spring', version: '2.2.1'
-  compile group: 'uk.gov.hmcts.reform', name: 'java-logging-appinsights', version: '2.2.1'
-
-  compile group: 'uk.gov.hmcts.reform', name: 'reform-api-standards', version: '0.3.0'
+//tasks.withType(JavaCompile) {
```
Comment (timja) [inline build.gradle]: ?

File: src/main/java/uk/gov/hmcts/reform/draftstore/data/model/Draft.java [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 import java.time.ZonedDateTime;
 
+/*
+ * This class should encapsulate functionality (OO) and not expose its internals "public final"
+ * We need to refactor this to follow good ood principles and practice.
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/draftstore/data/model/Draft.java]:
```suggestion
 * We need to refactor this to follow good oop principles and practice.
```

File: src/main/java/uk/gov/hmcts/reform/draftstore/domain/Draft.java [hunk 3]
```diff
@@ -4,6 +4,12 @@
 
 import java.time.ZonedDateTime;
 
+/*
+ * This class should encapsulate functionality (OO) and not expose its internals "public final"
+ * We need to refactor this to follow good ood principles and practice.
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/draftstore/domain/Draft.java]:
```suggestion
 * We need to refactor this to follow good oop principles and practice.
```

**Comment by KevinJCross** [comment]
> Update appinsights agent:
> 
> * [jar file](https://github.com/hmcts/draft-store/tree/master/lib)
> * [web conf](https://github.com/hmcts/draft-store/blob/master/web.config)
> 
> Review owasp suppressions. There are some resolved ones for sure
> 
> P.S. pmd rulesets will be deprecated in next major release. Should change to categories instead. But this is different matter. Can be addressed separately

@doncem  Ive updated the jar and file ... its not really part of this PR but Ive done it anyway. 

It does seem exceptionally weird this is not somehow included into the gradle file to download and include this dependency and the agent into the startup command.

**Comment by timja** [comment]
sounds like a java-logging update should be done 👍

**Pull request merged**